lms/djangoapps/oauth_dispatch/views.py

```diff
@@ -72,6 +82,71 @@ class AccessTokenView(_DispatchingView):
     dot_view = dot_views.TokenView
     dop_view = dop_views.AccessTokenView
 
+    @cached_property
+    def claim_handlers(self):
+        """ Returns a dictionary mapping scopes to methods that will add claims to the JWT payload. """
+
+        return {
+            'email': self._attach_email_claim,
+            'profile': self._attach_profile_claim
+        }
+
+    def dispatch(self, request, *args, **kwargs):
+        response = super(AccessTokenView, self).dispatch(request, *args, **kwargs)
+
+        if response.status_code == 200:
+            content = json.loads(response.content)
+            token = self._generate_jwt(request, content)
+            content['access_token'] = token
+            response.content = json.dumps(content)
+
+        return response
+
+    def _generate_jwt(self, request, content):
+        """ Returns a JWT access token. """
+
+        access_token = content['access_token']
+        access_token_obj = self.get_adapter(request).get_access_token(access_token)
+        user = access_token_obj.user
+        now = int(time())
+        content['token_type'] = 'JWT'
+
+        scope = content['scope']
+        payload = {
+            'iss': settings.JWT_AUTH['JWT_ISSUER'],
+            'aud': settings.JWT_AUTH['JWT_AUDIENCE'],
+            'jti': access_token,
+            'exp': now + content['expires_in'],
+            'iat': now,
+            'preferred_username': user.username,
+            'scope': scope,
+        }
+
+        scopes = scope.split(' ')
+
+        for scope in scopes:
+            handler = self.claim_handlers.get(scope)
+
+            if handler:
+                handler(payload, user)
+
+        secret = settings.JWT_AUTH['JWT_SECRET_KEY']
+        token = jwt.encode(payload, secret, algorithm=settings.JWT_AUTH['JWT_ALGORITHM'])
+
+        return token
+
+    def _attach_email_claim(self, payload, user):
+        """ Add the email claim details to the JWT payload. """
+        payload['email'] = user.email
+
+    def _attach_profile_claim(self, payload, user):
+        """ Add the profile claim details to the JWT payload. """
+        payload.update({
+            'family_name': user.last_name,
+            'name': user.get_full_name(),
+            'given_name': user.first_name,
+        })
+
```

Comment:
Yes, but that is easier said than done. django-oauth-toolkit limits access tokens to 255 characters, which is not long enough for JWTs.